src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
+  network: string | number;
+  /**
+   * Asset: "bitcoin", "ether" or { standard: "erc20", contract_address: "0x..." } for non-native tokens
+   */
+  // Not sure this is ideal, let's discuss and see what we can come up with
+  asset: string | { standard: "erc20"; contract_address: "0x..." };
+  /**
+   * Expiry: Expiry applying to this ledger, it can be an absolute or relative expiry, depending on the protocol and ledger
+   */
+  // We could have an object that allows the specification of the type of expiry here
+  expiry: number;
+}
+
+export enum EscrowStatus {
+  None = "NONE",
+  Initialized = "INITIALIZED",
+  Deployed = "DEPLOYED",
+  Funded = "FUNDED",
+  Redeemed = "REDEEMED",
+  Refunded = "REFUNDED",
+  IncorrectlyFunded = "INCORRECTLY_FUNDED"
+}
+
+/**
+ * The detailed description of the ledger state
+ */
+export interface LedgerState {
+  /**
+   * The address of the htlc.
+   */
+  htlc_location?: any;
+  /**
+   * The transaction ID of the deployment transaction.
+   */
+  deploy_tx?: string;
+  /**
+   * The transaction ID of the funding transaction.
+   */
+  fund_tx?: string;
+  /**
+   * The transaction ID of the redeem transaction.
+   */
+  redeem_tx?: string;
+  /**
+   * The transaction ID of the refund transaction.
+   */
+  refund_tx?: string;
+  /**
+   * The status of the escrow (HTLC or Invoice).
+   */
+  status: EscrowStatus;
+}
+
+// TODO: add `[k: string]: any;` in each section
+/**
+ * All the properties of a swap
+ */
+export interface FullSwapProperties {
+  /**
+   * The local id of the swap.
+   */
+  local_id: string;
+  /**
+   * The parameters of this swap.
+   */
+  // Decided to put everything together as now the protocols are tightly linked to the ledgers and assets
+  parameters: {
+    alpha: Parameters;
+    beta: Parameters;
+  };
+  /**
+   * The status this swap is currently in.
+   */
+  status: SwapStatus;
+  /**
+   * The data learned at creation of the swap.
+   */
+  created: {
+    /**
+     * The role in which you are participating in this swap.
+     */
+    role: "Alice" | "Bob";
+    /**
+     * The peer-id and hint address of the counterparty of this swap.
+     */
+    peer: Peer;
+    self_identities: {
```

Comment:
Do we have to tell the client what they told us?

Reply:
I think including both the `self` and `counterparty` will be useful for monitoring / dashboards. Also, the client could have switched wallets in between.

Reply:
I actually removed them when working on version 2, I have no re-added them as per Yos' request.

Reply:
When are we building monitoring or dashboards?

I don't think it is on the timeline anytime soon and even if it were/will be, then I would like to see a concrete usecase that requires `cnd` to share this information with the client.

I would assume that any wallet used by the user is tracking money by themselves and we don't need to tell it the addresses again.

Reply:
We can always add fields to this response later but once they are there, we need to maintain them and removing them is a backwards-incompatible change.

Just trying to make our own life easier :)